crates/router/src/types/domain/types.rs

```diff
@@ -94,7 +94,7 @@ impl<
         crypt_algo: V,
     ) -> CustomResult<Self, errors::CryptoError> {
         let encrypted = encrypted_data.into_inner();
-        let data = crypt_algo.decode_message(key, encrypted.clone())?;
+        let data = crypt_algo.decode_message(key, encrypted.clone().expose())?;
```

Comment:
don't expose if possible pass the secret directly. If need be you can take it up in a separate PR